### PR TITLE
Use Hamcrest assertions instead of JUnit in common/reactive (#1749)

### PR DESCRIPTION
**Pull request by Captain1653**
Part of #1749 

I'll do backport into helidon-2.x after merge.

I've already signed OCA. It's an automation problem.

I followed [these instructions](https://github.com/oracle/helidon/pull/1765#discussion_r423690868) 

- [x] [Backport into helidon-2.x](https://github.com/helidon-io/helidon/pull/4946)

### CODE REVIEW

File: common/reactive/src/test/java/io/helidon/common/reactive/SequentialSubscriberTest.java [hunk 1]
```diff
@@ -98,7 +99,7 @@ public void onComplete() {
         } catch (InterruptedException e) {
             executorService.shutdownNow();
         }
-        assertFalse(errorFound.isPresent(), () -> errorFound.get());
+        assertThat(errorFound.orElse(""), errorFound.isPresent(), is(false));
```
Comment (Captain1653) [inline common/reactive/src/test/java/io/helidon/common/reactive/SequentialSubscriberTest.java:102]: I don't like this approach with `Optional`, but if you need the error message...

Comment (tomas-langer) [reply common/reactive/src/test/java/io/helidon/common/reactive/SequentialSubscriberTest.java:102]: For Helidon 4 (`main` branch), we do have an optional matcher. For this version, we can keep this.

Alternative:
```java
assertThat(errorFound, is(Optional.empty()));
```

**Review by tomas-langer — APPROVED** [review]

**Pull request merged**